update.js

```diff
@@ -3,38 +3,42 @@
 const exec = require('child_process').execSync
 const fs = require('fs')
 
+const _ = require('lodash')
 const relative = require('require-relative')
 
 const extractDependency = require('./lib/extract-dependency')
 const updateShrinkwrap = require('./lib/update-shrinkwrap')
+const getValuesFromCI = require('./lib/get-values-from-ci')
 
 const pkg = relative('./package.json')
 
-const env = process.env
-
 module.exports = function update () {
   try {
     fs.readFileSync('./npm-shrinkwrap.json')
   } catch (e) {
     throw new Error('Without a shrinkwrap file present there is no need to run this script.')
   }
 
-  if (env.TRAVIS !== 'true') {
-    throw new Error('This script hast to run in an Travis CI environment')
+  const ciValues = getValuesFromCI()
+
+  if (_.isEmpty(ciValues)) {
+    throw new Error('This script must be run in a supported CI environment')
   }
 
-  if (env.TRAVIS_PULL_REQUEST !== 'false') {
+  const isPullRequest = _.get(ciValues, 'isPullRequest', false)
```

Comment:
Would be nice to have defaults applied in the lib file, so we can just use them here.
Also we don't need to repeat defaults if they're required in both upload and update.